src/js/component.js

```diff
@@ -1777,11 +1775,11 @@ class Component {
    *
    * @param {string} name
    *        The name of the `Component` to register.
-   *
-   * @param {Component} ComponentToRegister
+   * @template {Component} C
+   * @param {C} ComponentToRegister
    *        The `Component` class to register.
    *
-   * @return {Component}
+   * @return {C}
```

Comment:
The jsdoc output is bad here, has a `{C}` type that's not defined 
![image](https://github.com/videojs/video.js/assets/1676039/746bccfc-1c2b-4785-a448-31a15813f235)

whereas the `getComponent` you've modified below in video.js is displayed correctly, with a `{Class.<Component>}`
![image](https://github.com/videojs/video.js/assets/1676039/a6888081-3810-4eb8-bfc6-9175b40f01da)

Reply:
🙇 Thanks, fixed